lib/PuppeteerSharp.Tests/Puppeteer/PuppeteerLaunchTests.cs

```diff
@@ -38,5 +42,53 @@ public PuppeteerLaunchTests()
 
             await browser.CloseAsync();
         }
+
+        [Fact]
+        public async Task NetworkRedirectsShouldReportSecurityDetails()
+        {
+            var options = TestConstants.DefaultBrowserOptions();
+            options.IgnoreHTTPSErrors = true;
+
+            var browser = await PuppeteerSharp.Puppeteer.LaunchAsync(options, TestConstants.ChromiumRevision);
+            var page = await browser.NewPageAsync();
+            
+            var responses = new List<Response>();
+            page.ResponseCreated += (sender, e) => responses.Add(e.Response);
+
+            await page.GoToAsync(TestConstants.HttpsPrefix + "/plzredirect");
+            Assert.Equal(2, responses.Count);
```

Comment:
the test fails on this line, I commented out this line and it passed, not sure what the second response is supposed to be

Reply:
I tested that on Puppeteer, and Puppeteer is returning 2 responses, one with HTTP status 302 (redirect) and the other the empty page.
```JS
(async ()=>{
    const fs = require('fs');
    const path = require('path');
    
    const headless = (process.env.HEADLESS || 'true').trim().toLowerCase() === 'true';
    const slowMo = parseInt((process.env.SLOW_MO || '0').trim(), 10);
    const executablePath = process.env.CHROME;
    
    const defaultBrowserOptions = {
        executablePath,
        slowMo,
        headless,
        args: ['--no-sandbox', '--disable-dev-shm-usage']
    };

    const puppeteer = require("Puppeteer");

    const options = Object.assign({ignoreHTTPSErrors: true}, defaultBrowserOptions);
    const browser = await puppeteer.launch(options);
    const page = await browser.newPage();
    //httpsServer.setRedirect('/plzredirect', '/empty.html');
    const responses =  [];
    page.on('response', response => responses.push(response));
    await page.goto('http://127.0.0.1:8907/plzredirect');
    expect(responses.length).toBe(2);
    expect(responses[0].status()).toBe(302);
    const securityDetails = responses[0].securityDetails();
    expect(securityDetails.protocol()).toBe('TLS 1.2');
    await page.close();
    await browser.close();
})();
```

<img width="1283" alt="screen shot 2018-03-04 at 9 05 36 am" src="https://user-images.githubusercontent.com/2198466/36945389-193dc134-1f8c-11e8-9796-ad3f1b385790.png">


Something important @Meir017 is that we're working on Puppeteer v1.0, so it's important to read the test.js in that tag https://github.com/GoogleChrome/puppeteer/blob/v1.0.0/test/test.js#L109

I think this is the best approach because it would be a mess if we try to chase Puppeteer's guys on their master branch. So the approach would be implementing all tests we have on the v.1.0.0 tag and then move on to 1.1 and so on.

Though I think there is something off the response there, this is not on the test list on the issue. #3. But if you want to go and check what's missing (maybe some stupid mistake) go ahead and dig into this.